spot-contracts/contracts/PerpetualTranche.sol

```diff
@@ -495,7 +530,9 @@ contract PerpetualTranche is
         (int256 reserveFee, uint256 protocolFee) = feeStrategy.computeBurnFees(perpAmtBurnt);
 
         // updates the mature tranche balance
-        _updateMatureTrancheBalance((_matureTrancheBalance * (perpSupply - perpAmtBurnt)) / perpSupply);
+        _updateMatureTrancheBalance(
+            MathUpgradeable.mulDiv(_matureTrancheBalance, (perpSupply - perpAmtBurnt), perpSupply)
```

Comment:
Could you use a `using` statement, then just:
`_matureTrancheBalance.mulDiv(perpSupply - perpAmtBurnt, perpSupply)`?
Am I remembering that syntax correctly?

Reply:
yea we can do that